### DISCUSSION TITLE
Initialising queryCache

### DISCUSSION THREAD

**Discussion by prashanthacharyam** [TanStack/query#472] [General] [answered]
I have a api call which gets me all the initial data ( like statues, dropdown values which rarely change ).
I can use queryCache.setQueryData(KEY, data) to initialise the cache. 
But without a staleTime the query is called again if used in some other hook.
Is there a way to initialise the cache with staleTime?

**Comment by ryanmk54** [comment]
Stale time can be configured at the [global and query level](https://github.com/tannerlinsley/react-query#caching--invalidation).

> By default, query results become stale immediately after a successful fetch. This can be configured using the staleTime option at both the global and query-level.

You can either set it at the [global level with queryConfigProvider](https://github.com/tannerlinsley/react-query#reactqueryconfigprovider) or [per query when you call useQuery](https://github.com/tannerlinsley/react-query#usequery)

**Comment by ryanmk54** [comment]
You may be interested in this feature request
https://github.com/tannerlinsley/react-query/discussions/451

**Reply by prashanthacharyam** [reply]
I was looking into something like **queryCache.setQueryData(queryKey, updater, { staleTime: Infinity})** 

The internal code has it **cache.setQueryData = (queryKey, updater, { exact, ...config } = {})**, but not exposed outside i guess. Have not checked the source code.

**Comment by prashanthacharyam** [comment, marked as answer]
I now have a parent component which loads the data from the server. It passes the data to child component which calls the hooks and sets the initialData field.
